container/controller/controller/runtime.py

```diff
@@ -145,15 +166,11 @@ def delete(self, domname):
                                       "-l", "domain=%s" % domname])
 
     def connection_uri(self, domname):
-        data = kubectl(["get", "service", "-ojson",
-                        "-l", "domain=%s" % domname])
-        # FIXME directly fetch for instance
-        parsed = json.loads(data)
-        clusterip = jsonpath("items[0].spec.clusterIP",
-                             parsed)[0].value
-        port = jsonpath("items[0].spec.ports[?(name='libvirt')].port",
-                        parsed)[0].value
-        return "qemu+tcp://%s:%s/system" % (clusterip, port)
+        pods = get_rc_pod_names("compute-rc-%s" % (domname))
```

Comment:
The pod name should not be constructed, rather a selector should be used, i.e. `-l domain=$DOMNAME`